faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -919,6 +919,26 @@ class ClientSpec
       identifyR.to[Boolean].get shouldBe true
       }
 
+  it should "test CurrentToken with internal token" in {
+    val createR = client.query(Create(Collection("spells"), Obj("credentials" -> Obj("password" -> "abcdefg")))).futureValue
+
+    // Login
+    val loginR = client.query(Login(createR(RefField), Obj("password" -> "abcdefg"))).futureValue
+    val secret = loginR(SecretField).get
+    val tokenRef = loginR(RefField).get
+
+    // CurrentToken
+    val currentToken = client.sessionWith(secret)(_.query(CurrentToken())).futureValue
+    currentToken.to[RefV].get shouldBe tokenRef
+  }
+
+  it should "test CurrentToken with internal key" in {
+    val clientKey = adminClient.query(CreateKey(Obj("role" -> "client"))).futureValue
+    val clientKeyClient = FaunaClient(secret = clientKey(SecretField).get, endpoint = config("root_url"))
+    val currentToken = clientKeyClient.query(CurrentToken()).futureValue
+    currentToken.to[RefV].get shouldBe clientKey(RefField).get
+  }
+
   it should "create session client" in {
```

Comment:
@nathaniel-may, I know it's not as straightforward as the other cases, but just to confirm, are we supposed to test the `CurrentToken` function upon being logged in with an external provider here in the driver? :)

> 3. Auth'ed with external provider --> return object with all the JWT claims

Reply:
because it requires an external auth provider I'd say it's more of a integration test. We'll QA this before the release for the driver, but core already has good tests for this.

Reply:
Alright, fair enough. That sounds good to me!